### ISSUE TITLE
In Terminal.php line 131:    exec(): Unable to fork [stty 2>&1]

### ISSUE THREAD

**Issue by switch263** [alexjustesen/speedtest-tracker#400]
**Describe the bug**
"Unable to fork" error on clean startup. I'm able to run all other containers as desired without issue. ulimit values within container appear appropriate, and the system has ample free resources. 

**To Reproduce**
Deploy docker container, observe failure of any tests to run, and errors in docker logs. Docker-compose entry:
```
  speedtest-tracker:
    container_name: speedtest-tracker
    ports:
        - '8765:80'
        - '8443:443'
    environment:
        - PUID=1001
        - PGID=1001
    volumes:
        - '/data/speedtest-tracker:/config'
    image: 'ghcr.io/alexjustesen/speedtest-tracker:latest'
    restart: unless-stopped
```

**Expected behavior**
Application to start normally

**Environment (please complete the following information):**
 - OS: Ubuntu / docker
-  Architecture: amd64
 - Browser chrome
 - Version  v0.11.1

**Screenshots**
N/A - nothing of value shows in the UI. It appears normal but does not ever get any results data, since forking fails.

```
**Logs**
💪  Starting the queue worker...
[2023-02-09 18:41:01] production.ERROR: exec(): Unable to fork [stty 2>&1] {"exception":"[object] (ErrorException(code: 0): exec(): Unable to fork [stty 2>&1] at /var/www/html/vendor/symfony/console/Terminal.php:131)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(266): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /var/www/html/vendor/symfony/console/Terminal.php(131): exec()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(123): Symfony\\Component\\Console\\Terminal::hasSttyAvailable()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(663): Illuminate\\Container\\BoundMethod::call()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Illuminate\\Container\\Container->call()
#10 /var/www/html/vendor/symfony/console/Command/Command.php(312): Illuminate\\Console\\Command->execute()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(152): Symfony\\Component\\Console\\Command\\Command->run()
#12 /var/www/html/vendor/symfony/console/Application.php(1022): Illuminate\\Console\\Command->run()
#13 /var/www/html/vendor/symfony/console/Application.php(314): Symfony\\Component\\Console\\Application->doRunCommand()
#14 /var/www/html/vendor/symfony/console/Application.php(168): Symfony\\Component\\Console\\Application->doRun()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(102): Symfony\\Component\\Console\\Application->run()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(155): Illuminate\\Console\\Application->run()
#17 /var/www/html/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#18 {main}
"}

In Terminal.php line 131:

  exec(): Unable to fork [stty 2>&1]
```

**Additional context**
N/A

**Comment by alexjustesen** [comment]
Can you try deleting the image and pulling the `latest`?

**Comment by brum74** [comment]
I have the same problem on a fresh setup.  I'm running docker in a debian vm.

**Comment by alexjustesen** [comment]
Double check you `PUID` and `PGID` values as read/write errors can occur if these aren't set correctly. https://docs.speedtest-tracker.dev/getting-started/installation#required-configuration

**Comment by brum74** [comment]
I verified my UID=1000 and GID=1000 and attached more of the logging which shows it should be set.
```
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

‍♂️ Checking for Laravel automations...
 SSL_MODE has been set to MIXED, setting the web server to work in HTTP + HTTPS...
 Linking the storage...
‍♂️ An SSL key was not detected, so I'll generate a self-signed SSL certificate pair for you...
.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*...+...+....+...+...+...........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

   INFO  The [public/storage] link has been connected to [storage/app/public].  

  Configuring Speedtest Tracker...
✅  Database exists
  Creating database symlink
✅  Environment file exists
  Creating symlinks to config and log files
✅  App key exists
  Fixing app path file permissions
  Building the cache...
.+.....................+..+.......+...........+.+...+........+.........+.+........+.+.....+...+.+......+.....+.+..+...+.........+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.....+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.........+.+......+.........+............+...+...........+...+.+..+.......+........+.......+..+....+..+.......+........+....+...........+....+........+...+..........+........+.+......+..............+...+......+......+................+...+......+........+...+...+....+.....+..........+..+.+..+.......+......+.....+...+................+...+......+.........+......+.........+..+......+...+....+............+...+..+...+.......+...............+......+.....+....+...+......+..+...+.........+...............+.............+........+.+.....+....+.....+.+.....+.+.....+.......+...........+....+..+.............+.....+.+...........+....+..+....+......+.........+........+......+......+....+......+...+.....+.......+...........+...+....+..+...+.+..+.........+....+.....+.+...+...............+..+.........+......+...+.......+........+..........+......+..+......................+......+......+...+...+............+...+..+......+............+...+......+......+...+.+......+...+............+.................+.+............+...+..+....+........+.+.....+.+.....+...+.+..+....[02-Mar-2023 11:08:01] NOTICE: fpm is running, pid 102
[02-Mar-2023 11:08:01] NOTICE: ready to handle connections
[02-Mar-2023 11:08:01] NOTICE: systemd monitor interval set to 10000ms
......+.....+...+.......+........+.......+............+...........+....+...+.....+...+.........+.+...............+.....+.+...+...+.....+....+.....+.+.....+.........+..........+.........+..+....+...+...........+.+.....+.+..+.......+.........+...........+......+...+.......+........+......+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
  Migrating the database...
✅  All set, starting Speedtest Tracker container...
⏰  Starting the cron service...
  Starting the queue worker...
[2023-03-02 16:08:01] production.ERROR: exec(): Unable to fork [stty 2>&1] {"exception":"[object] (ErrorException(code: 0): exec(): Unable to fork [stty 2>&1] at /var/www/html/vendor/symfony/console/Terminal.php:131)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(266): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /var/www/html/vendor/symfony/console/Terminal.php(131): exec()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(123): Symfony\\Component\\Console\\Terminal::hasSttyAvailable()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(661): Illuminate\\Container\\BoundMethod::call()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call()
#10 /var/www/html/vendor/symfony/console/Command/Command.php(312): Illuminate\\Console\\Command->execute()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(153): Symfony\\Component\\Console\\Command\\Command->run()
#12 /var/www/html/vendor/symfony/console/Application.php(1022): Illuminate\\Console\\Command->run()
#13 /var/www/html/vendor/symfony/console/Application.php(314): Symfony\\Component\\Console\\Application->doRunCommand()

#14 /var/www/html/vendor/symfony/console/Application.php(168): Symfony\\Component\\Console\\Application->doRun()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(102): Symfony\\Component\\Console\\Application->run()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(155): Illuminate\\Console\\Application->run()
#17 /var/www/html/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#18 {main}
"} 
In Terminal.php line 131:
                                      
  exec(): Unable to fork [stty 2>&1]  
```

This is my compose:
```
  speedtest-local:
    container_name: speedtest-local
    image: ghcr.io/alexjustesen/speedtest-tracker:latest
    restart: unless-stopped
    ports:
      - 8766:80
      - 8767:443
    environment:
      - TZ=America/New_York
      - PGID=1000
      - PUID=1000
    volumes:
      - ./speedtest/data/config:/config
```

**Comment by snown** [comment]
I'm seeing the same issue. Brand new install on docker

**Comment by jimueller** [comment]
Seeing the same on Docker Swarm, verified PUID / PGID match.  This app looks really great so I'm determined to get it working, despite not being a PHP dev.

I thought it might be nice to run strace to see if I could get more info on what's actually failing. After observing the processes I thought following the s6-supervise queue-worker and it's child threads would be the most promising.

`strace -p <QUEUE WORKER PID> -f -Z`

Need to track these down a bit more, but decided to post in case anything jumps out to anyone.  Looks like it's possibly having trouble finding bash or the run script.

```
[pid 558709] +++ exited with 1 +++
[pid 558708] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=5053, si_uid=0, si_status=1, si_utime=0, si_stime=0} ---
[pid 558708] execve("/command/bash", ["bash", "./run"], 0x7ffc97bae9f8 /* 34 vars */) = -1 ENOENT (No such file or directory)
[pid 558708] execve("/usr/local/sbin/bash", ["bash", "./run"], 0x7ffc97bae9f8 /* 34 vars */) = -1 ENOENT (No such file or directory)
[pid 558708] execve("/usr/local/bin/bash", ["bash", "./run"], 0x7ffc97bae9f8 /* 34 vars */) = -1 ENOENT (No such file or directory)
[pid 558708] execve("/usr/sbin/bash", ["bash", "./run"], 0x7ffc97bae9f8 /* 34 vars */) = -1 ENOENT (No such file or directory)
[pid 558708] arch_prctl(0x3001 /* ARCH_??? */, 0x7fffe7e03b80) = -1 EINVAL (Invalid argument)
[pid 558708] access("/etc/ld.so.preload", R_OK) = -1 ENOENT (No such file or directory)
[pid 558708] openat(AT_FDCWD, "/dev/tty", O_RDWR|O_NONBLOCK) = -1 ENXIO (No such device or address)
[pid 558708] ioctl(0, TCGETS, 0x7fffe7e03a00) = -1 ENOTTY (Inappropriate ioctl for device)
[pid 558708] ioctl(-1, TIOCGPGRP, 0x7fffe7e03a74) = -1 EBADF (Bad file descriptor)
[pid 558708] newfstatat(AT_FDCWD, "/command/bash", 0x7fffe7e036b0, 0) = -1 ENOENT (No such file or directory)
[pid 558708] newfstatat(AT_FDCWD, "/usr/local/sbin/bash", 0x7fffe7e036b0, 0) = -1 ENOENT (No such file or directory)
[pid 558708] newfstatat(AT_FDCWD, "/usr/local/bin/bash", 0x7fffe7e036b0, 0) = -1 ENOENT (No such file or directory)
[pid 558708] newfstatat(AT_FDCWD, "/usr/sbin/bash", 0x7fffe7e036b0, 0) = -1 ENOENT (No such file or directory)
[pid 558708] connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
[pid 558708] connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
[pid 558708] ioctl(2, TIOCGPGRP, 0x7fffe7e03914) = -1 ENOTTY (Inappropriate ioctl for device)
[pid 558708] ioctl(2, TIOCGPGRP, 0x7fffe7e038e4) = -1 ENOTTY (Inappropriate ioctl for device)
[pid 558708] ioctl(3, TCGETS, 0x7fffe7e03a10) = -1 ENOTTY (Inappropriate ioctl for device)
[pid 558708] fcntl(255, F_GETFD)        = -1 EBADF (Bad file descriptor)
```

**Comment by alexjustesen** [comment]
I'm releasing `v0.11.3` which only includes updates to dependencies, if anyone could test to see if this resolves the issues above as I'm not able to replicate them.

**Comment by alexjustesen** [comment]
Are you building the image locally or using the published ones?

**Comment by switch263** [comment]
I used the published one of the latest build. I can try building it locally later today if that might help.

**Comment by alexjustesen** [comment]
Give it a whack just in case there is a problem with my builds for whatever reason

**Comment by switch263** [comment]
I was able to build the image successfully on a Ubuntu 22.04 box but am still unable to run the container, although it does give a different error now when I try to run it. On both this box and a Ubuntu 18 box I had sat around but not in use, when using the published builds, I still see the same 'unable to fork' error.

```
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

🏃‍♂️ Checking for Laravel automations...
🔐 Linking the storage...
🔒 SSL_MODE has been set to MIXED, setting the web server to work in HTTP + HTTPS...
🏃‍♂️ An SSL key was not detected, so I'll generate a self-signed SSL certificate pair for you...
........+....+...+...+.........+..+.+..+......+.+.....+......+.........+..........+.....+.........+....+..+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*........+..+.+......+...............+.....+.+.........+........+...+...+..........+.....+....+...+...+......+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*............+....+.....+.+...+.....+.....................+.............+.....................+...........+...+...+...+.+...+..+.........+......+.+...+.....+......+....+......+........+......+............+.........+.+.......................+......+...+.......+..................+........+.......+..+...+......+.+...............+...+.....+.+...........+.........+.......................................+.+.........+..+.........+.+......+...+..+...+.+..............+.........+.......+...+............+......+...+.........+......+...........+.........+.........+...+.........+...+.+.................+............+.......+.....+.+...+.....+......+.+..+...............+.+.....+.........+......+......+....+.....+.+.....+....+...........+...............+.+...............+...+..+....+.........+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

   INFO  The [public/storage] link has been connected to [storage/app/public].

🐇  Configuring Speedtest Tracker...
🙄  Database file not found, creating...
🔗  Creating database symlink
✅  Environment file exists
🔗  Creating symlinks to config and log files
✅  App key exists
🔒  Fixing app path file permissions
💰  Building the cache...
....+.+..+....+...+...+..+..........+...+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+......+...+.....+...+.........+....+..+...+............+.......+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*......+............+............+...+...+..+.......+.....+...+.........+...+......+....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
[06-Mar-2023 19:53:10] NOTICE: fpm is running, pid 102
[06-Mar-2023 19:53:10] NOTICE: ready to handle connections
[06-Mar-2023 19:53:10] NOTICE: systemd monitor interval set to 10000ms
🚛  Migrating the database...
✅  All set, starting Speedtest Tracker container...
⏰  Starting the cron service...
💪  Starting the queue worker...
/run/s6/basedir/scripts/rc.init: 67: restart:: not found
[06-Mar-2023 19:53:11] NOTICE: Terminating ...
[06-Mar-2023 19:53:11] NOTICE: exiting, bye-bye!
```

**Comment by alexjustesen** [comment]
@switch263 can you add `DEBUG_OUTPUT=true` to your environment vars when creating the container, this will enable a more verbose error output. Example below...

```
docker run -d --name speedtest-tracker \
    -p 8080:80 \
    -p 8443:443 \
    -e PUID=1000 \
    -e PGID=1000 \
    -e DEBUG_OUTPUT=true \
    -v /path/to/_data:/config \
    speedtest-tracker
```

**Comment by switch263** [comment]
@alexjustesen dropped the logs into a gist here https://gist.github.com/switch263/99ab8e597a2cdef9166a1d05a2b3b074

Hope these can help!

**Comment by alexjustesen** [comment]
@switch263 one more level of outputs, can you also add `S6_VERBOSITY='all'` and share a new gist?

**Comment by somera** [comment]
Here ...
[speedtest-tracker_v0.11.4_debug.log](https://github.com/alexjustesen/speedtest-tracker/files/10912316/speedtest-tracker_v0.11.4_debug.log)

**Comment by alexjustesen** [comment]
Best I can tell it looks like fpm isn't starting for you which is causing the container to stop. I'll dig into it a little more...

**Comment by alexjustesen** [comment]
I had a typo in the env var, can you use  `S6_VERBOSITY=5`?

**Comment by alexjustesen** [comment]
`v0.11.6` #454 has been released in the hopes it resolves this issue. If it has please comment and I'll close the issue.

**Comment by snown** [comment]
I updated my docker image pointing to `ajustesen/speedtest-tracker` this morning, and I am still seeing the same error. If it helps my console output is here:

https://gist.github.com/snown/1bf9dc85dcb15f52f03007e7abd7c79b

**Comment by switch263** [comment]
At some point over the last couple days since I last had time to work on my homelab stack, my stack restarted and pulled the latest build and this is now seemingly working as expected. I haven't had time to do extensive testing yet but general functionality is normal for me - the application successfully starts, I'm able to log in, run tests, etc. Since others have chimed in with the same problem I'll leave closing this to @alexjustesen but I'm able to use the tool as intended now. Thanks for all the hard work! :)

**Comment by alexjustesen** [comment]
Going to close this out for now as the author is all set. Going to continue to improve the startup process though

**Comment by brum74** [comment]
I know this is closed but just wanted to comment that this still happens for me and hoping your future startup improvements resolves it.

**Comment by jimueller** [comment]
In case this helps for anyone else, it does seem 0.11.6 fixed it for me as well, but I did do a couple of other things.

- updated to 0.11.6 and app would not start - complaining of unable to write sqlite - had not had that happen before.
- Switched from using docker volume to bind mount within the PUID user's home directory

So, not entirely sure what the fix was, something in 0.11.6 either fixed it or exposed the volume permission issue that may have been the root cause.  It is strange though because prior to 0.11.6 I had no issues with the volume.

**Comment by alexdelprete** [comment]
> So, not entirely sure what the fix was, something in 0.11.6 either fixed it or exposed the volume permission issue that may have been the root cause. It is strange though because prior to 0.11.6 I had no issues with the volume.

Good catch. I bet my $0.02 on the volume's permissions, and suggest @alexjustesen to modify the recommended docker installation using bind mounts.

With all my containers, I use bind mounts and not volumes and probably that's the reason why I'm always unable to reproduce these strange errors many users are reporting.

**Comment by sjg** [comment]
Have stared a discussion about this as I'm having the same issues on a fresh install. 

[Discussion #541](https://github.com/alexjustesen/speedtest-tracker/discussions/541)
